rewrite-gradle/src/main/java/org/openrewrite/gradle/RemoveEnableFeaturePreview.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.gradle;
+
+import java.util.List;
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.jetbrains.annotations.NotNull;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.groovy.GroovyIsoVisitor;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.J.Literal;
+import org.openrewrite.java.tree.J.MethodInvocation;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class RemoveEnableFeaturePreview extends Recipe {
+
+  @Option(displayName = "The feature preview name",
+      description = "The name of the feature preview to remove.",
+      example = "ONE_LOCKFILE_PER_PROJECT"
+  )
+  public String previewFeatureName;
+
+  @Override
+  public @NotNull String getDisplayName() {
+    return "Remove an enabled Gradle preview feature";
+  }
+
+  @Override
+  public @NotNull String getDescription() {
+    return "Remove an enabled Gradle preview feature from `settings.gradle`.";
+  }
+
+  @Override
+  public TreeVisitor<?, ExecutionContext> getVisitor() {
+    return Preconditions.check(
+        new IsSettingsGradle<>(),
+        new RemoveEnableFeaturePreviewVisitor());
+  }
+
+  public class RemoveEnableFeaturePreviewVisitor extends GroovyIsoVisitor<ExecutionContext> {
+
+    @Override
+    public MethodInvocation visitMethodInvocation(
+        MethodInvocation method, @NotNull ExecutionContext executionContext) {
+
+      if (!"enableFeaturePreview".equals(method.getSimpleName())) {
```

Comment:
I was using a method matcher, and it passed tests fine, but I manually ran this recipe against a few local gradle projects and it worked on 1 but on another it didn't match at all. The one it didn't work on, I debugged the LST and found that the method had a methodType of null for some reason. I dont' know if it's a bug in the parser or not, but I changed the logic here to just check the method name, check that there's exactly 1 argument, and check that the argument matches the `@Option` passed in.

Reply:
It being flaky is a bit curious to me. My likely suspect though is that in the cases where it is not working, the type is missing from the Groovy AST -- coming out of the Groovy Compiler -- prior to it being mapped to the OpenRewrite Groovy LST.